keras/engine/training.py

```diff
@@ -1893,9 +1893,9 @@ def fit_generator(self, generator,
                 non picklable arguments to the generator
                 as they can't be passed
                 easily to children processes.
-            shuffle: Whether to shuffle the data at the beginning of each
-                epoch. Only used with instances of `Sequence` (
-                keras.utils.Sequence).
+            shuffle: Whether to shuffle the order of the batches at 
```

Comment:
The inserted ```order of``` is less precise than the original. It is the batches that are shuffled, not their order. The order is randomised.

Reply:
@MatthiasWinkelmann I think that contradicts what @fchollet said, but I may have confused him.

I think I've sorted (shuffled?) it out.

The only place `fit_generator` uses its `shuffle` argument is where it passes it into the constructor for `OrderedEnqueuer`:
https://github.com/fchollet/keras/blob/master/keras/engine/training.py#L2002

The only place `OrderedEnqueuer` uses its constructor's `shuffle` argument is where it looks like it's shuffling the list of indices which it will use to access the underlying sequence:
https://github.com/fchollet/keras/blob/9c79e01a61b371820d04e9cc52487fc0eb5d0312/keras/utils/data_utils.py#L490

If that's doing what I think it is, then it's shuffling the order of the batches, not the data within each mini-batch, though that could be a good addition for parity with `model.fit(..., shuffle="batch")`.